tests/test.py

```diff
@@ -7246,6 +7249,117 @@ def testRotatedKeysAcraBlock(self):
+        sa.Column('token_email', sa.Text, nullable=False, default=''),
+    )
+
+    def setUp(self):
+        super().setUp()
+        self.engine1.execute(self.encryptor_table_join.delete())
+
+    def testSearchWithJoinedTable(self):
+        context = self.get_context_data()
+        search_term = context['searchable']
+
+        # Insert searchable data and some additional different rows
+        self.insertRow(context)
+        self.insertDifferentRows(context, count=5)
+
+        # Insert the same data into encryptor_table_join table
+        self.insertRow(context, query=self.encryptor_table_join.insert())
+        self.insertDifferentRows(context, count=5, query=self.encryptor_table_join.insert())
+
+        rows = self.executeSelect2(
+            sa.select(
+                self.encryptor_table.c.number,
+                self.encryptor_table_join.c.searchable,
+                self.encryptor_table_join.c.default_client_id,
+                self.encryptor_table_join.c.raw_data,
+                self.encryptor_table_join.c.specified_client_id,
+                self.encryptor_table_join.c.zone_id,
+                self.encryptor_table_join.c.empty,
+            ).
+            where(self.encryptor_table.c.searchable == sa.bindparam('searchable')).
+            join(self.encryptor_table_join, self.encryptor_table_join.c.searchable==sa.bindparam('searchable')),
+            {'searchable': search_term})
+        self.assertEqual(len(rows), 1)
+        row = rows[0]
+
+        self.assertEqual(row[0], context['number'])
+        self.assertEqual(row[1], search_term)
+
+        # should be decrypted
+        self.assertEqual(row['default_client_id'], context['default_client_id'])
+        # should be as is
+        self.assertEqual(row['raw_data'], context['raw_data'])
+        # other data should be encrypted
+        self.assertNotEqual(row['specified_client_id'], context['specified_client_id'])
+        self.assertNotEqual(row['zone_id'], context['zone_id'])
+        self.assertEqual(row['empty'], b'')
+
+        # test with invalid search_term
+        rows = self.executeSelect2(
+            sa.select(
+                self.encryptor_table.c.number,
+                self.encryptor_table_join.c.searchable,
+            ).
+            where(self.encryptor_table.c.searchable == sa.bindparam('searchable')).
+            join(self.encryptor_table_join, self.encryptor_table_join.c.searchable==sa.bindparam('searchable')),
+            {'searchable': get_pregenerated_random_data().encode('utf-8')})
+        self.assertEqual(len(rows), 0)
+
+        # test join with on table1.searchable = table2.searchable
+        rows = self.executeSelect2(
+            sa.select(self.encryptor_table.c.number, self.encryptor_table_join.c.searchable).
+            where(self.encryptor_table.c.searchable == sa.bindparam('searchable')).
+            join(self.encryptor_table_join, self.encryptor_table.c.searchable==self.encryptor_table_join.c.searchable),
```

Comment:
can we test raw join without `where`? or as I understand, we test same successful case as above?

Reply:
Added separate query without `where`